### PR TITLE
Added slide expiration date

### PR DESCRIPTION
**Pull request by PrestaShark**
Sometimes we forget about expired slides on our store. This will help to prevent unwanted old slider banners.

expiration

Fixes https://github.com/PrestaShop/PrestaShop/issues/12991

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Hi there,

Thanks for this PR. I see that you're altering the database with this new version. Would you mind adding an upgrade find for merchants upgrading their module from a previous version? (See https://devdocs.prestashop.com/1.7/modules/creation/enabling-auto-update/ for more details)

However, I see how you manage these expiration date on the configuration page, but where do you apply the filter on the front office?

Regards

File: ps_imageslider.php [hunk 1]
```diff
@@ -188,6 +189,7 @@ protected function createTables()
               `id_homeslider_slides` int(10) unsigned NOT NULL AUTO_INCREMENT,
               `position` int(10) unsigned NOT NULL DEFAULT \'0\',
               `active` tinyint(1) unsigned NOT NULL DEFAULT \'0\',
+              `expiration` datetime NOT NULL,
```
Comment (jolelievre) [inline ps_imageslider.php]: I think we shouldn't force an expiration date, some of the slides may stay here indefinitely

File: ps_imageslider.php [hunk 2]
```diff
@@ -447,6 +449,7 @@ protected function _postProcess()
             $slide->position = (int)Tools::getValue('position');
             /* Sets active */
             $slide->active = (int)Tools::getValue('active_slide');
+            $slide->expiration = Tools::getValue('expiration');
```
Comment (jolelievre) [inline ps_imageslider.php]: If `expiration` query parameter is empty set `null` in database

**Comment by SimonGrn** [comment]
Since we got no response from you, I close this PR. Feel free to reopen it if new information pops up.

Thank you.

**Pull request closed without merging**